README.md

```diff
@@ -16,84 +22,11 @@ Or [`npm`](https://www.npmjs.com/package/@amsterdam/use-promise):
-    return <p>Something went wrong fetching the user information: ${result.error.message}</p>
-  }
-
-  return <p>You're looking at the user information of ${result.value.firstName}!<p>
-}
-
-export default UserDetails
-```
-
-Let's break this example down to the essentials, starting with the `usePromise` hook.
-
-```ts
-const result = usePromise(() => fetchUser(userId), [userId])
-```
-
-First off, we pass a function as the first argument. This is the factory function that is responsible for creating our promise.
-
-This function is called when our component is mounted, or if the dependencies passed in as the second argument change. This is essentially the same as the [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) hook.
-
-The `usePromise` function returns a special object that contains the `status` of the Promise and some other fields. Using this `status` field we can determine what to show the user.
+## Guides
+- [Getting Started](./guides/getting-started.md)
+- [Handling Side-effects](./guides/side-effects.md)
+- [Cancelling Fetch Requests](./guides/cancelling-requests.md)
 
-```tsx
-if (result.status === PromiseStatus.Pending) {
-  return <p>Fetching user information&hellip;</p>
-}
-
-if (results.status === PromiseStatus.Rejected) {
-  return <p>Something went wrong fetching the user information: ${result.error.message}</p>
-}
-```
-
-Here we are handling the different statuses of the Promise to inform the user about what is going on. When the status is `Pending` we show some loading state, and if it is `Rejected` we inform the user that something went wrong. If the status is `Rejected` we can also retrieve the `error` value from the Promise to find out exactly what went wrong.
-
-There are 3 possible statuses:
-- `Pending` - The Promise is in-flight, meaning it was not yet resolved or rejected.
-- `Rejected` - The Promise is rejected, something went wrong.
-- `Resolved` - The Promise is resolved, all went well.
-
-On the last line of our component we can now safely assume that the status is `Resolved` as we already checked all other possible statuses. This means that we can now also access the `value` property to get the resolved value of the Promise.
-
-```tsx
-return <p>You're looking at the user information of ${result.value.firstName}!<p>
-```
+## Compatibility
 
-That's it, you're now the master of Promises! 💪
+This library will work in every environment (Node or browser) that supports [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController), [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal), [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) and [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function). If you need to support an environment that does not have these features make sure to include the appropriate polyfills and transformations.
```

Comment:
Perhaps refer to babel for transformations / polyfills? As this is one of the most used compiler

Reply:
I'd like to keep this out for now as there should be plenty of resources around for users on how to do this. Most tooling such a Create React App and Next will do this for you automatically.